strava/client.py

```diff
@@ -261,18 +261,24 @@ def get_segment(self, segment_id):
         path = f'segments/{segment_id}'
         return self._dispatcher('get', path)
 
-    def get_segment_efforts(self, segment_id, per_page=50, limit=None):
+    def get_segment_efforts(self, segment_id, per_page=50, limit=None, start_date=None, end_date=None):
         """
         Return all segment's efforts from activities of the authenticated user.
 
         See docs: http://developers.strava.com/docs/reference/#api-SegmentEfforts-getEffortsBySegmentId
 
         :param segment_id [int]: Segment id.
+        :param start_date [Date]: Segment start date.
+        :param end_date [Date]: Segment end date.
         :param per_page [int]: page size.
         :param limit [int]: maximum number of activities to fetch.
         """
 
-        path = f'segments/{segment_id}/all_efforts'
+        path = f'segment_efforts?segment_id={segment_id}'
+
+        if start_date and end_date:
+            path += f'&start_date_local={start_date}&end_date_local={end_date}'
+
```

Comment:
Since we're using `requests` lib as HTTP client we should keep the pattern to pass querystring as named arguments into `BatchIterator` as did bellow. ie. 
`BatchIterator(fetcher, segment_id=segment_id, per_page=per_page, limit=limit, start_date_local=start_dt, end_date_local=end_dt)`

Also, you should add `start_date_local` and `end_date_local` only when they are not  `None`

The last thing is Strava API ask to send the datetime string in format ISO 8601, then you should convert the datetime object into ISO 8601 string, I'd try the following format `%Y-%m-%dT%H:%M:%SZ`. for example:

```python
if start_dt:
    start_date_local = start_dt.strftime('%Y-%m-%dT%H:%M:%SZ')
```